$OpenBSD$

reorder includes. From getifaddr(3) manual:

"If both <net/if.h> and <ifaddrs.h> are being 
included, <net/if.h> must be included before <ifaddrs.h>"

MERGED UPSTREAM 
https://github.com/bitcoin/bitcoin/pull/4119

--- src/compat.h.orig	Tue Apr  8 12:04:06 2014
+++ src/compat.h	Fri May  2 19:55:15 2014
@@ -28,16 +28,16 @@
 #include <windows.h>
 #include <ws2tcpip.h>
 #else
+#include <sys/fcntl.h>
+#include <sys/mman.h>
+#include <sys/socket.h>
 #include <sys/types.h>
+#include <net/if.h>
+#include <netinet/in.h>
 #include <arpa/inet.h>
 #include <ifaddrs.h>
 #include <limits.h>
-#include <net/if.h>
 #include <netdb.h>
-#include <netinet/in.h>
-#include <sys/fcntl.h>
-#include <sys/mman.h>
-#include <sys/socket.h>
 #include <unistd.h>
 #endif
 
